### ISSUE TITLE
Error: Expected type

### ISSUE THREAD

**Issue by Manav361** [llvm/llvm-project#95767]
#### Problem Description:
When running the following command to optimize `OPT.ll` by adding O3 optimization:

```sh
opt -O3 -S -o OPT_O3.ll OPT.ll 
```

an error occurs indicating an error in expected type:

```
OPT.ll:12:63: warning: ptr type is only supported in -opaque-pointers mode
@om_external_constant_data_constant_733_opt = internal global ptr null, align 16
                                                              ^
opt: OPT.ll:12:63: error: expected type
@om_external_constant_data_constant_733_opt = internal global ptr null, align 16
                                                              ^                                                        ^
```
                                                                        

I have attached the OPT.ll file to reproduce the error by running the above mentioned command.
[OPT.ll](https://drive.google.com/file/d/1BKlToipv8MOvJ4X3m5IYnNIAI_S3TYDa/view?usp=sharing) . I got this OPT.ll file from mlir-translate command.

**Comment by tgymnich** [comment]
Try running opt with `--opaque-pointers`.

**Comment by Manav361** [comment]
> Try running opt with `--opaque-pointers`.

I tried doing that but it throws an unexpected attribute error as follows with the following command.

```sh
opt --opaque-pointers -O3 -S -o OPT_O3.ll OPT.ll
```

an error occurs indicating an error in expected type:


```
opt: OPT.ll:156495:57: error: unterminated attribute group
attributes #0 = { nocallback nofree nounwind willreturn memory(argmem: readwrite) }
                                                        ^
 ```

**Comment by tgymnich** [comment]
mlir-translate might be targeting a different llvm version. Can you post the version of opt you are using?
